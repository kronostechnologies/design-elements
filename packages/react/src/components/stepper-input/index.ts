export { StepperInput } from './stepper-input';
export type { StepperInputProps } from './stepper-input';
