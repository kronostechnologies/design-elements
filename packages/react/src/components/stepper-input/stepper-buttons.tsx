import React, { ReactElement } from 'react';
import styled, { css } from 'styled-components';

import { Icon } from '../icon/icon';

const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    height: 32px;
    width: 23px;
`;

const ButtonStyles = css`
    align-items: center;
    background-color: ${({ theme }) => theme.greys.white};
    border: 1px solid ${({ theme }) => theme.greys['dark-grey']};
    border-left: none;
    color: ${({ theme }) => theme.greys['dark-grey']};
    cursor: pointer;
    display: flex;
    justify-content: center;
    padding: 0 var(--spacing-half);

    &:focus {
        outline: none;
    }

    &:hover {
        background-color: ${({ theme }) => theme.greys.grey};
    }

    &:disabled {
        background-color: ${({ theme }) => theme.greys['light-grey']};
        border-color: ${({ theme }) => theme.greys.grey};
        color: ${({ theme }) => theme.greys['mid-grey']};
        cursor: default;
    }
`;

const IncrementButton = styled.button`
    ${ButtonStyles}

    border-bottom: none;
    border-radius: 0 var(--border-radius) 0 0;
    height: calc(50% - 1px);
`;

const DecrementButton = styled.button`
    ${ButtonStyles}

    border-radius: 0 0 var(--border-radius) 0;
    height: calc(50% + 1px);
`;

interface StepperButtonsProps {
    disabled?: boolean;
    onIncrement?(): void;
    onDecrement?(): void;
}

export function StepperButtons({ disabled, onIncrement, onDecrement }: StepperButtonsProps): ReactElement {
    return (
        <Wrapper>
            <IncrementButton
                data-testid="stepper-button-increment"
                type="button"
                onClick={onIncrement}
                disabled={disabled}
            >
                <Icon name="chevronUp" size="16" />
            </IncrementButton>
            <DecrementButton
                data-testid="stepper-button-decrement"
                type="button"
                onClick={onDecrement}
                disabled={disabled}
            >
                <Icon name="chevronDown" size="16" />
            </DecrementButton>
        </Wrapper>
    );
}
