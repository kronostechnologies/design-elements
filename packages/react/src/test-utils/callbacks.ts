export function doNothing(): void {
    // do nothing
}
