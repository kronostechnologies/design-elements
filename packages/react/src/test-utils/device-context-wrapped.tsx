import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
