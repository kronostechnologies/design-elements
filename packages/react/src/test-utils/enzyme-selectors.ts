import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
