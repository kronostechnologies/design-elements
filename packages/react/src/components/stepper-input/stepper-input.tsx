import {
    ChangeEvent,
    DetailedHTMLProps,
    InputHTMLAttributes,
    RefObject,
    useMemo,
    useRef,
    VoidFunctionComponent,
} from 'react';
import styled from 'styled-components';
import { useTranslation } from '../../i18n/use-translation';
import { Theme } from '../../themes';
import { v4 as uuid } from '../../utils/uuid';
import { DeviceContextProps, useDeviceContext } from '../device-context-provider/device-context-provider';
import { FieldContainer } from '../field-container/field-container';
import { responsiveInputsStyle } from '../text-input/styles/inputs';
import { TooltipProps } from '../tooltip/tooltip';
import { StepperButtons } from './stepper-buttons';

const Wrapper = styled.div`
    display: flex;
    max-width: fit-content;
`;

interface StyledInputProps {
    device: DeviceContextProps;
    theme: Theme;
}

const StyledInput = styled.input<StyledInputProps>`
    ${responsiveInputsStyle}

    border-radius: ${({ device }) => (device.isMobile ? 'var(--border-radius)' : 'var(--border-radius) 0 0 var(--border-radius)')};
    height: ${({ device }) => (device.isMobile ? 40 : 32)}px;
    width: ${({ device }) => (device.isMobile ? 164 : 56)}px;
    z-index: 1;

    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
        -webkit-appearance: none; /* stylelint-disable-line property-no-vendor-prefix */
        margin: 0;
    }

    &[type=number] {
        -moz-appearance: textfield; /* stylelint-disable-line property-no-vendor-prefix */
    }
`;

type PartialStepperInputProps = Pick<DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
    'disabled' | 'onFocus' | 'onBlur' | 'step'>;

type Value = undefined | number | null;

interface StepperInputProps extends PartialStepperInputProps {
    defaultValue?: number;
    hint?: string;
    id?: string;
    label?: string;
    max?: number;
    min?: number;
    noMargin?: boolean;
    tooltip?: TooltipProps;
    valid?: boolean;
    validationErrorMessage?: string;
    value?: Value;

    onChange?(value: Value): void
}

function triggerChangeEventOnRef(ref: RefObject<HTMLInputElement>): void {
    // Rationale for using dispatchEvent: https://github.com/kronostechnologies/design-elements/pull/180#discussion_r556050899
    ref.current?.dispatchEvent(new Event('change', { bubbles: true }));
}

export const StepperInput: VoidFunctionComponent<StepperInputProps> = ({
    defaultValue,
    disabled,
    hint,
    id,
    label,
    max,
    min,
    noMargin,
    step,
    tooltip,
    valid = true,
    validationErrorMessage,
    value,
    onBlur,
    onChange,
    onFocus,
}) => {
    const inputRef = useRef<HTMLInputElement>(null);
    const { t } = useTranslation('stepper-input');
    const device = useDeviceContext();
    const fieldId = useMemo(() => id || uuid(), [id]);

    function handleIncrement(): void {
        const valueBefore = Number(inputRef.current?.value);
        inputRef.current?.stepUp();
        const valueAfter = Number(inputRef.current?.value);

        if (valueBefore !== valueAfter) {
            triggerChangeEventOnRef(inputRef);
        }
    }

    function handleDecrement(): void {
        const valueBefore = Number(inputRef.current?.value);
        inputRef.current?.stepDown();
        const valueAfter = Number(inputRef.current?.value);

        if (valueBefore !== valueAfter) {
            triggerChangeEventOnRef(inputRef);
        }
    }

    const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
        const inputValue: string = event.target.value;
        if (inputValue === '') {
            onChange?.(null);
        } else {
            const currentValue = Number(inputValue);
            onChange?.(currentValue);
        }
    };

    return (
        <FieldContainer
            fieldId={fieldId}
            hint={hint}
            label={label}
            tooltip={tooltip}
            noMargin={noMargin}
            valid={valid}
            validationErrorMessage={validationErrorMessage || t('validationErrorMessage')}
        >
            <Wrapper>
                <StyledInput
                    data-testid="stepper-input"
                    defaultValue={defaultValue}
                    device={device}
                    disabled={disabled}
                    id={fieldId}
                    max={max}
                    min={min}
                    name="points"
                    onChange={handleChange}
                    onBlur={onBlur}
                    onFocus={onFocus}
                    ref={inputRef}
                    step={step}
                    type="number"
                    value={value === null ? '' : value}
                />
                {!device.isMobile && (
                    <StepperButtons
                        disabled={disabled}
                        onDecrement={handleDecrement}
                        onIncrement={handleIncrement}
                    />
                )}
            </Wrapper>
        </FieldContainer>
    );
};
