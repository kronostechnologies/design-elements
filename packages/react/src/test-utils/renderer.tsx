import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
