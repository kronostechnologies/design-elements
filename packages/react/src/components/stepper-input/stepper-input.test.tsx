import { shallow } from 'enzyme';
import React from 'react';

import { getByTestId } from '@design-elements/test-utils/enzyme-selectors';
import { renderWithProviders } from '@design-elements/test-utils/renderer';
import { StepperInput } from './stepper-input';

jest.mock('uuid/v4');

describe('Stepper input', () => {
    test('onChange callback should be called when input value changes', () => {
        const callback = jest.fn();
        const wrapper = shallow(<StepperInput onChange={callback} />);

        getByTestId(wrapper, 'stepper-input').simulate('change', { value: 3 });

        expect(callback).toHaveBeenCalledTimes(1);
    });

    test('onBlur callback should be called when input is blurred', () => {
        const callback = jest.fn();
        const wrapper = shallow(<StepperInput onBlur={callback} />);

        getByTestId(wrapper, 'stepper-input').simulate('blur');

        expect(callback).toHaveBeenCalledTimes(1);
    });

    test('onFocus callback should be called when input is focused', () => {
        const callback = jest.fn();
        const wrapper = shallow(<StepperInput onFocus={callback} />);

        getByTestId(wrapper, 'stepper-input').simulate('focus');

        expect(callback).toHaveBeenCalledTimes(1);
    });

    test('matches snapshot', () => {
        const tree = renderWithProviders(<StepperInput label="test" />);

        expect(tree).toMatchSnapshot();
    });

    test('matches snapshot (mobile)', () => {
        const tree = renderWithProviders(<StepperInput label="test" />, 'mobile');

        expect(tree).toMatchSnapshot();
    });

    test('matches snapshot (invalid)', () => {
        const tree = renderWithProviders(<StepperInput label="test" valid={false} />);

        expect(tree).toMatchSnapshot();
    });

    test('matches snapshot (disabled)', () => {
        const tree = renderWithProviders(<StepperInput label="test" disabled />);

        expect(tree).toMatchSnapshot();
    });
});
