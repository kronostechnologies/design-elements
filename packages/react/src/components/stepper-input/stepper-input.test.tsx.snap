// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper input matches snapshot (disabled) 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0:focus {
  border-color: #006296;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 32px;
  width: 23px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-bottom: none;
  border-radius: 0 var(--border-radius) 0 0;
  height: calc(50% - 1px);
}

.c6:focus {
  outline: none;
}

.c6:hover {
  background-color: #DBDEE1;
}

.c6:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-radius: 0 0 var(--border-radius) 0;
  height: calc(50% + 1px);
}

.c7:focus {
  outline: none;
}

.c7:hover {
  background-color: #DBDEE1;
}

.c7:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c4 {
  background: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-sizing: border-box;
  color: #000000;
  font-family: inherit;
  font-size: 0.875rem;
  -webkit-letter-spacing: 0.015rem;
  -moz-letter-spacing: 0.015rem;
  -ms-letter-spacing: 0.015rem;
  letter-spacing: 0.015rem;
  line-height: 1.5rem;
  margin: 0;
  outline: none;
  padding: var(--spacing-half) var(--spacing-1x);
  width: 100%;
  border-radius: var(--border-radius) 0 0 var(--border-radius);
  height: 32px;
  width: 56px;
  z-index: 1;
}

.c4::-webkit-input-placeholder {
  color: #60666E;
}

.c4::-moz-placeholder {
  color: #60666E;
}

.c4:-ms-input-placeholder {
  color: #60666E;
}

.c4::placeholder {
  color: #60666E;
}

.c4:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
}

.c4:disabled,
.c4:disabled::-webkit-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::-moz-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled:-ms-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::placeholder {
  color: #B7BBC2;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #84C6EA;
}

.c4::-webkit-outer-spin-button,
.c4::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c4[type=number] {
  -moz-appearance: textfield;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
  >
    test
  </label>
  <div
    class="c3"
  >
    <input
      class="c4"
      data-testid="stepper-input"
      disabled=""
      name="points"
      type="number"
    />
    <div
      class="c5"
    >
      <button
        aria-label="Increase number"
        class="c6"
        data-testid="stepper-button-increment"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
      <button
        aria-label="Decrease number"
        class="c7"
        data-testid="stepper-button-decrement"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Stepper input matches snapshot (invalid) 1`] = `
.c3 {
  color: #CD2C23;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
}

.c5 {
  margin-left: var(--spacing-base);
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #CD2C23;
}

.c0:focus {
  border-color: #CD2C23;
}

.c0 > :nth-child(2) {
  margin-bottom: var(--spacing-half);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 32px;
  width: 23px;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-bottom: none;
  border-radius: 0 var(--border-radius) 0 0;
  height: calc(50% - 1px);
}

.c9:focus {
  outline: none;
}

.c9:hover {
  background-color: #DBDEE1;
}

.c9:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-radius: 0 0 var(--border-radius) 0;
  height: calc(50% + 1px);
}

.c10:focus {
  outline: none;
}

.c10:hover {
  background-color: #DBDEE1;
}

.c10:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c7 {
  background: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-sizing: border-box;
  color: #000000;
  font-family: inherit;
  font-size: 0.875rem;
  -webkit-letter-spacing: 0.015rem;
  -moz-letter-spacing: 0.015rem;
  -ms-letter-spacing: 0.015rem;
  letter-spacing: 0.015rem;
  line-height: 1.5rem;
  margin: 0;
  outline: none;
  padding: var(--spacing-half) var(--spacing-1x);
  width: 100%;
  border-radius: var(--border-radius) 0 0 var(--border-radius);
  height: 32px;
  width: 56px;
  z-index: 1;
}

.c7::-webkit-input-placeholder {
  color: #60666E;
}

.c7::-moz-placeholder {
  color: #60666E;
}

.c7:-ms-input-placeholder {
  color: #60666E;
}

.c7::placeholder {
  color: #60666E;
}

.c7:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
}

.c7:disabled,
.c7:disabled::-webkit-input-placeholder {
  color: #B7BBC2;
}

.c7:disabled,
.c7:disabled::-moz-placeholder {
  color: #B7BBC2;
}

.c7:disabled,
.c7:disabled:-ms-input-placeholder {
  color: #B7BBC2;
}

.c7:disabled,
.c7:disabled::placeholder {
  color: #B7BBC2;
}

.c7:focus {
  outline: none;
}

.c7:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #84C6EA;
}

.c7::-webkit-outer-spin-button,
.c7::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c7[type=number] {
  -moz-appearance: textfield;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
  >
    test
  </label>
  <div
    aria-live="polite"
    class="c3"
    id="undefined_invalid"
    role="alert"
  >
    <div
      class="c4"
    >
      <svg
        color="currentColor"
        focusable="false"
        height="16"
        width="16"
      />
    </div>
    <span
      class="c5"
    >
      Invalid number
    </span>
  </div>
  <div
    class="c6"
  >
    <input
      class="c7"
      data-testid="stepper-input"
      name="points"
      type="number"
    />
    <div
      class="c8"
    >
      <button
        aria-label="Increase number"
        class="c9"
        data-testid="stepper-button-increment"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
      <button
        aria-label="Decrease number"
        class="c10"
        data-testid="stepper-button-decrement"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Stepper input matches snapshot (mobile) 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.5rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0:focus {
  border-color: #006296;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c4 {
  background: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-sizing: border-box;
  color: #000000;
  font-family: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: 0.02875rem;
  -moz-letter-spacing: 0.02875rem;
  -ms-letter-spacing: 0.02875rem;
  letter-spacing: 0.02875rem;
  line-height: 1.5rem;
  margin: 0;
  outline: none;
  padding: var(--spacing-1x);
  width: 100%;
  border-radius: var(--border-radius);
  height: 40px;
  width: 164px;
  z-index: 1;
}

.c4::-webkit-input-placeholder {
  color: #60666E;
}

.c4::-moz-placeholder {
  color: #60666E;
}

.c4:-ms-input-placeholder {
  color: #60666E;
}

.c4::placeholder {
  color: #60666E;
}

.c4:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
}

.c4:disabled,
.c4:disabled::-webkit-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::-moz-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled:-ms-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::placeholder {
  color: #B7BBC2;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #84C6EA;
}

.c4::-webkit-outer-spin-button,
.c4::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c4[type=number] {
  -moz-appearance: textfield;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
  >
    test
  </label>
  <div
    class="c3"
  >
    <input
      class="c4"
      data-testid="stepper-input"
      name="points"
      type="number"
    />
  </div>
</div>
`;

exports[`Stepper input matches snapshot 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  line-height: 1.25rem;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

input + .c1 {
  margin-left: var(--spacing-half);
}

.c0 {
  margin: 0 0 var(--spacing-3x);
}

.c0 input,
.c0 select,
.c0 textarea {
  border-color: #60666E;
}

.c0:focus {
  border-color: #006296;
}

.c0 > :nth-child(1) {
  margin-bottom: var(--spacing-half);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 32px;
  width: 23px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-bottom: none;
  border-radius: 0 var(--border-radius) 0 0;
  height: calc(50% - 1px);
}

.c6:focus {
  outline: none;
}

.c6:hover {
  background-color: #DBDEE1;
}

.c6:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-left: none;
  color: #60666E;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 var(--spacing-half);
  border-radius: 0 0 var(--border-radius) 0;
  height: calc(50% + 1px);
}

.c7:focus {
  outline: none;
}

.c7:hover {
  background-color: #DBDEE1;
}

.c7:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
  color: #B7BBC2;
  cursor: default;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c4 {
  background: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-sizing: border-box;
  color: #000000;
  font-family: inherit;
  font-size: 0.875rem;
  -webkit-letter-spacing: 0.015rem;
  -moz-letter-spacing: 0.015rem;
  -ms-letter-spacing: 0.015rem;
  letter-spacing: 0.015rem;
  line-height: 1.5rem;
  margin: 0;
  outline: none;
  padding: var(--spacing-half) var(--spacing-1x);
  width: 100%;
  border-radius: var(--border-radius) 0 0 var(--border-radius);
  height: 32px;
  width: 56px;
  z-index: 1;
}

.c4::-webkit-input-placeholder {
  color: #60666E;
}

.c4::-moz-placeholder {
  color: #60666E;
}

.c4:-ms-input-placeholder {
  color: #60666E;
}

.c4::placeholder {
  color: #60666E;
}

.c4:disabled {
  background-color: #F1F2F2;
  border-color: #DBDEE1;
}

.c4:disabled,
.c4:disabled::-webkit-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::-moz-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled:-ms-input-placeholder {
  color: #B7BBC2;
}

.c4:disabled,
.c4:disabled::placeholder {
  color: #B7BBC2;
}

.c4:focus {
  outline: none;
}

.c4:focus {
  outline: none;
  border-color: #006296;
  box-shadow: 0 0 0 2px #84C6EA;
}

.c4::-webkit-outer-spin-button,
.c4::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c4[type=number] {
  -moz-appearance: textfield;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
  >
    test
  </label>
  <div
    class="c3"
  >
    <input
      class="c4"
      data-testid="stepper-input"
      name="points"
      type="number"
    />
    <div
      class="c5"
    >
      <button
        aria-label="Increase number"
        class="c6"
        data-testid="stepper-button-increment"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
      <button
        aria-label="Decrease number"
        class="c7"
        data-testid="stepper-button-decrement"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="currentColor"
          focusable="false"
          height="16"
          width="16"
        />
      </button>
    </div>
  </div>
</div>
`;
